proto/token.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+import "permissions.proto";
+
+option go_package = "github.com/momentohq/client-sdk-go;client_sdk_go";
+option java_multiple_files = true;
+option java_package = "momento.auth";
+
+package token;
+
+service Token {
+  rpc GenerateTempAuthToken (_GenerateTempAuthTokenRequest) returns (stream _GenerateTempAuthTokenResponse) {}
```

Comment:
```suggestion
  rpc GetTemporaryAuthToken (_GetTemporaryAuthTokenRequest) returns (_GetTemporaryAuthTokenResponse) {}
```

* I think you can just use "get" instead of "generate" to mean the same thing.
* We should prefer clear, full words like "temporary" over needless abbreviations like "temp."
* If you think you need a server-side streaming response, let's chat on slack. I'm guessing that was just a mistaken copypaste?

Reply:
- I think you can just use "get" instead of "generate" to mean the same thing.
  -  I like that
- We should prefer clear, full words like "temporary" over needless abbreviations like "temp."
  - Yes, this name also happens to be temporary, this pr is in effort to get more of a discussion/decision on the API name
- If you think you need a server-side streaming response, let's chat on slack. I'm guessing that was just a mistaken copypaste?
  - Yep, def a copypaste miss
  
I'll clean these up once name is settled

Reply:
i think we aligned in slack that we are not going to switch to `Get`, we will keep `Generate` for consistency with existing API. Jury might still be out on `Temporary` vs other names